lib/gc-dialects/OneDNNGraph/OneDNNGraphOps.cpp

```diff
@@ -0,0 +1,184 @@
+//===- OneDNNGraphOps.h - OneDNN input dialect ops --------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "gc-dialects/OneDNNGraph/OneDNNGraphOps.h"
+#include "gc-dialects/OneDNNGraph/OneDNNGraphDialect.h"
+#include "mlir/IR/OpImplementation.h"
+#include "llvm/Support/Debug.h"
+
+#define GET_OP_CLASSES
+#include "gc-dialects/OneDNNGraph/OneDNNGraphOps.cpp.inc"
+
+namespace mlir {
+namespace onednn_graph {
+
+// https://github.com/onnx/onnx/blob/main/docs/Broadcasting.md
+template <typename ShapeRange>
+static LogicalResult inferBroadcastShape(
```

Comment:
Do we need it for the dialect definition or can add later?

Reply:
inferBroadcastShape will be deleted next, while properly crediting the author.